Change images of KinD configuration files to be compatible with v0.14.0

NickLoukas opened the pull request:
As requested here https://github.com/kubeflow/manifests/pull/2237#issuecomment-1171430252 we're moving all the KinD configuration files in one place and make sure that they all use the images from v0.14.0.

kimwnasptd commented on tests/gh-actions/kind-cluster-1-22.yaml:
```diff
@@ -1,3 +1,4 @@
+# This configuration file should work with KinD v0.14.0
```
Since you changed the name from `kind-1-22.yaml` to `kind-cluster-1-22.yaml` we'll need to update the actions that use this config.

NickLoukas replied:
Fixed it.

NickLoukas commented:
@kimwnasptd I addressed the review comments above and rebased on top of the recent changes in master to resolve all the conflicts. Please take a look.
